library/src/main/scala/scala/scalajs/js/special/package.scala

```diff
@@ -92,6 +92,29 @@ package object special {
   def delete(obj: scala.Any, key: scala.Any): Unit =
     throw new java.lang.Error("stub")
 
+  /** Enumerates the keys of an object.
+   *
+   *  This method is the exact equivalent of the `for (a in o) {}` statement of
+   *  JavaScript. A call of the form
+   *  {{{
+   *  forin(obj)(f)
+   *  }}}
+   *  corresponds to the JavaScript statement
+   *  {{{
+   *  for (const x in obj) {
+   *    f(x)
+   *  }
+   *  }}}
+   *
+   *  `for..in` loops exhibit performance cliffs in most JavaScript engines,
+   *  which means their performance is very dependent on their surroundings.
+   *  Therefore, it is recommended to avoid this method and use
+   *  [[js.Any.ObjectCompanionOps.properties js.Object.properties]] instead, if
```

Comment:
So just to double check: `js.Object.properties` calls this method internally, but just makes sure it does so properly, right?

Reply:
Yes, that's right.